### PR TITLE
Remove mmx register support on x86

### PR DESCRIPTION
**Pull request by BradleyWood**
See #6273.

Signed-off-by: BradleyWood <bradley.wood@ibm.com>

### CODE REVIEW

**Comment by BradleyWood** [comment]
FYI, @0xdaryl

**Comment by 0xdaryl** [comment]
Jenkins build all

File: compiler/x/codegen/OMRMachine.cpp [hunk 1]
```diff
@@ -270,15 +270,15 @@ OMR::X86::Machine::findBestFreeGPRegister(TR::Instruction   *currentInstruction,
          first = TR::RealRegister::FirstGPR;
          last  = TR::RealRegister::Last8BitGPR;
          break;
+      case TR_DoubleWordReg:
+#if defined(TR_TARGET_X86)
```
Comment (0xdaryl) [inline compiler/x/codegen/OMRMachine.cpp:274]: This #ifdef could be removed.  This is code in the x86 code generator.

File: compiler/x/codegen/OMRMachine.cpp [hunk 2]
```diff
@@ -452,15 +452,15 @@ TR::RealRegister *OMR::X86::Machine::freeBestGPRegister(TR::Instruction
...
```
Comment (0xdaryl) [inline compiler/x/codegen/OMRMachine.cpp:456]: This #ifdef could be removed. This is code in the x86 code generator.

File: compiler/x/codegen/OMRMachine.cpp (hunk 1 above)
Comment (BradleyWood) [reply compiler/x/codegen/OMRMachine.cpp:274]: Isn't this code common to 32 and 64 bit?

Comment (0xdaryl) [reply compiler/x/codegen/OMRMachine.cpp:274]: Yes, but `TR_TARGET_X86` applies to both.  If you want to make it specific to 32-bit, for example, you also need to use `defined (TR_TARGET_32BIT)` or `defined(TR_TARGET_64BIT)` for 64-bit.

**Comment by BradleyWood** [comment]
Jenkins build xlinux,x32linux

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**